thoth/package_extract/image.py

```diff
@@ -281,6 +288,72 @@ def _run_apt_cache_show(
     return result
 
 
+def _get_lib_dir_symbols(root_dir, zeros=16):
+    to_ret = set({})
+    zero_string = "0" * zeros
+    command = f"nm -D {root_dir}/*so* | grep '{zero_string} A'"
+    out = subprocess.Popen(command, stdout=subprocess.PIPE, shell=True).stdout.read().decode("utf-8")
+    lines = out.split('\n')
+    for line in lines:
+        columns = line.split(' ')
+        if len(columns) > 2:
+            to_ret.add(columns[2])
+    return to_ret
+
+
+def _root_lib_symbols(path: str):
+    to_ret = set()
+    if os.path.exists(os.path.join(path, "usr/lib64")):
+        to_ret |= _get_lib_dir_symbols(os.path.join(path, "usr/lib64"))
+        to_ret |= _get_lib_dir_symbols(os.path.join(path, "lib64"))
+    elif os.path.exists(os.path.join(path, "usr/lib32")):
+        to_ret |= _get_lib_dir_symbols(os.path.join(path, "usr/lib32"), zeros=8)
+        to_ret |= _get_lib_dir_symbols(os.path.join(path, "lib32"), zeros=8)
+    elif os.path.exists(os.path.join(path, "usr/lib")):
+        to_ret |= _get_lib_dir_symbols(os.path.join(path, "usr/lib"), zeros=8)
+        to_ret |= _get_lib_dir_symbols(os.path.join(path, "lib"), zeros=8)
+    else:
+        raise FileNotFoundError("No usr libraries were found")
+
+    return to_ret
+
+
+def _ld_config_symbols(path: str):
+    to_ret = set()
+    os.chdir(os.path.join(path, "etc/"))
```

Comment:
Use https://github.com/thoth-station/common/blob/f6cb5fdd0e65a28d8000f3671990a751dcef6a44/thoth/common/helpers.py#L33 instead so the location is preserved across function calls.